### PR TITLE
Teleportation tutorial

### PR DESCRIPTION
**Pull request by timmysilv**
**Title:**
Quantum Teleportation

**Summary:**
Now there will be a PennyLane tutorial on how to perform quantum teleportation 😎 This is largely a copy-paste of Olivia's [lesson from the codebook](https://codebook.xanadu.ai/I.15). Some notes:
- I don't have pics ready... was wondering if I could get some assistance on that front 😄 as of now, I've just copied from the codebook
- I added classical control wires to circuit-drawing so I could showcase a variety of PennyLane features to their fullest (mid-circuit measurement stuff and draw_mpl), so I'm not using `qml.draw_mpl` with the "sketch" option to draw things (releasing that on Monday... so it won't draw nicely until then)

**Relevant references:**
Codebook (linked above)

**Related GitHub Issues:**
I added circuit-drawing in PennyLaneAI/pennylane#4228, and you can see what it looks like in the attached image.

With the `sketch` style (barrier location and StatePrep label are outdated here):
image

With the new `pennylane` style:
image

### CODE REVIEW

File: demonstrations/tutorial_teleportation.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+many useful applications across the entire field. These principles include (but
+are not limited to): the no-cloning theorem, quantum entanglement, and the
+principle of deferred measurement. Let's dive in!
+
+Suppose there are two researchers named Alice and Bob, and Alice wants to send
+her quantum state to Bob. The quantum teleportation protocol enables Alice to
+do exactly this in a very elegant manner. An overview of the protocol can be seen
+here:
+
+.. figure:: ../demonstrations/teleportation/teleportation-4part.svg
```
Comment (timmysilv) [inline demonstrations/tutorial_teleportation.py:27]: this should probably be a cooler image 😄

File: demonstrations/tutorial_teleportation.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+**no-cloning theorem**. The proof is surprisingly straightforward. Suppose we
+would like to design a circuit (unitary) :math:`U` that can perform the following
+action:
+
+.. math::
+
+    \begin{align*}
+    U(\vert \psi\rangle \otimes \vert s\rangle ) &= \vert \psi\rangle \otimes \vert \psi\rangle \\
+    U(\vert \varphi\rangle \otimes \vert s\rangle ) &= \vert \varphi \rangle \otimes \vert \varphi \rangle
+    \tag{1}\end{align*}
```
Comment (ludmilaasb) [inline demonstrations/tutorial_teleportation.py:49]:
```suggestion
    \end{align*}
```

Comment (ludmilaasb) [reply demonstrations/tutorial_teleportation.py:49]: I didn't find any reference to this equation through the text

File: demonstrations/tutorial_teleportation.py [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+where :math:`\vert \psi\rangle` and :math:`\vert \varphi\rangle` are arbitrary
+single-qubit states, and :math:`\vert s \rangle` is some arbitrary starting state.
+We will now prove that no such :math:`U` exists! First, note that the inner product
```
Comment (ludmilaasb) [inline demonstrations/tutorial_teleportation.py:53]:
```suggestion
We will now prove that no such :math:`U` exists! First, let's take the inner product
```

File: demonstrations/tutorial_teleportation.py [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+her quantum state to Bob. The quantum teleportation protocol enables Alice to
+do exactly this in a very elegant manner. An overview of the protocol can be seen
+here:
+
+.. figure:: ../demonstrations/teleportation/teleportation-4part.svg
+    :align: center
+    :width: 75%
+
+Disclaimer: It should be noted that it is only the quantum *information* being
+teleported, and not a physical particle.
```
Comment (ludmilaasb) [inline demonstrations/tutorial_teleportation.py:32]: This is fine but it could be more naturally integrated to the text with a briefly description of the protocol. See above.

File: demonstrations/tutorial_teleportation.py (hunk 2 above)
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py]: Good point - I originally just kept the tags from the codebook page. In my recent commit, I removed all tags that aren't referenced later (and re-indexed the remaining ones accordingly)

File: demonstrations/tutorial_teleportation.py [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+
+where :math:`\vert \psi\rangle` and :math:`\vert \varphi\rangle` are arbitrary
+single-qubit states, and :math:`\vert s \rangle` is some arbitrary starting state.
+We will now prove that no such :math:`U` exists! First, note that the inner product
+of the left-hand-sides of the two equations is :math:`\langle \psi \vert \varphi \rangle`.
+Next, note that the inner product of the right-hand-sides of the two equations is :math:`(\langle \psi \vert \varphi \rangle)^2`.
+These inner products must be equal, and they are only equal if they are a value that
+squares to itself. The only valid values for the inner product then are 1 and 0. But
+if the inner product is 1, the states are the same; on the other hand, if the inner
+product is 0, the states are orthogonal. Therefore, we can't clone arbitrary states!
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py]: I kinda like how it's done in the tutorial itself, since the math is pretty straightforward/quick and the reader can follow along without having to leave our page, and hopefully learn something new! if they already know it, they'll probably just skip this part anyway 😄

**Review by ludmilaasb — CHANGES_REQUESTED** [review]
Hello! 

Thanks for preparing this very nice demo!

The pace is good, also the content. I suggested some minor changes to make the text more fluid. 

Good job!

File: demonstrations/tutorial_teleportation.py [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+Now that we know we can't arbitrarily copy states, we return to the task of
+teleporting them. Teleportation relies on Alice and Bob having access to
+shared entanglement. The protocol can be divided into roughly four parts. We'll
+go through each of them in turn.
+
+.. figure:: ../demonstrations/teleportation/teleportation-4part.svg
+    :align: center
+    :width: 75%
+
+"""
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:78]: I think I'd wanna re-draw the whole picture with trendy PennyLane graphics style, then at the top of each sub-section, re-post the picture but only highlighting the component being discussed. I can take a crack at it but I am not much of an illustrator...

File: demonstrations/tutorial_teleportation.py [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+##############################################################################
+#
+# 4. Measurement
+# ``````````````
+#
+# The last step of the protocol involves applying two controlled operations from
+# Alice's qubits to Bob, a controlled-:math:`Z`, and a :math:`CNOT`, followed by a
+# measurement. But why exactly are we doing this before the measurement? In the
+# previous step, we already performed a basis rotation back to the computational
+# basis, so shouldn't we be good to go? Not quite, but almost!
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:214]: totally agreed, and I actually had a hard time with this, because I wanted to explain only traditional stuff first too. I ended up writing it in this way to match the 4-part diagram (which has CZ and CNOT), but I'm only now realizing that we could just change that diagram to show classical measurements!

File: demonstrations/tutorial_teleportation.py [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+@qml.qnode(dev)
+def teleport(state):
+    state_preparation(state)
+    entangle_qubits()
+    basis_rotation()
+    measure_and_update()
+    return qml.state()
+
+
+_ = qml.draw_mpl(teleport, style="sketch")(state)
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:292]: do you mean how it went from the classical double-wires to the regular CNOT gate? if so - that was the intention:) perhaps I can make it clearer that it's what I'm trying to demonstrate? (if a QNode binds a quantum function to a device that does not support mid-circuit measurements, it applies the deferred measurement principle)

File: demonstrations/tutorial_teleportation.py [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+# 2. :math:`\vert\cdot\rangle_a`, Alice's auxiliary (or "ancilla") qubit that
+#    she will entangle with Bob's qubit for communication purposes
+# 3. :math:`\vert \cdot\rangle_B`, Bob's qubit that will receive the teleported
+#    state
+#
+# Together, their starting state is:
+#
+# .. math::
+#
+#     \vert 0\rangle_A \vert 0\rangle_a \vert 0\rangle_B.\tag{2}
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py]: I mention it before the basis-change math (`Expanding out the terms from (1) (and removing the subscripts for brevity)`), but I'll make it more obvious.

File: demonstrations/tutorial_teleportation.py [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+#     \vert \psi_+\rangle &= \frac{1}{\sqrt{2}} \left( \vert 00\rangle + \vert 11\rangle \right), \\
+#     \vert \psi_-\rangle &= \frac{1}{\sqrt{2}} \left( \vert 00\rangle - \vert 11\rangle \right), \\
+#     \vert \phi_+\rangle &= \frac{1}{\sqrt{2}} \left( \vert 01\rangle + \vert 10\rangle \right), \\
+#     \vert \phi_-\rangle &= \frac{1}{\sqrt{2}} \left( \vert 01\rangle - \vert 10\rangle \right).
+#     \tag{5}\end{align*}
+#
+# After the basis transform, if we observe the first two qubits to be in the state
+# :math:`\vert 00\rangle`, this would correspond to the outcome :math:`\vert \psi_+\rangle` in
+# the Bell basis, :math:`\vert 11\rangle` would correspond to :math:`\vert \phi_-\rangle`,
+# etc. Let's perform this change of basis, one step at a time.
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:165]: I think it's better here actually - it's not related to wire $B$ being in the desired $\vert \psi\rangle$ state, but rather it's just generally describing a change of basis. I think moving it down might confuse the reader

File: demonstrations/tutorial_teleportation.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:318]: oh my, slipped right by me! the `defer_measurements` function must be making something like `qml.ctrl(qml.PauliX)`, which _should_ get turned into a `CNOT` but doesn't. This is a bug imo:
```python
def c():
    m0 = qml.measure(0)
    qml.cond(m0, qml.PauliX)(1)

mid_circuit = qml.tape.make_qscript(c)()
deferred = qml.defer_measurements(mid_circuit)
print(deferred.operations)
# [Controlled(PauliX(wires=[1]), control_wires=[0])]
```

we should register `qml.ctrl` to make `CNOT` gates when possible

[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:318]: opened PennyLaneAI/pennylane#4339

File: demonstrations/tutorial_teleportation.py [hunk 11]
```diff
@@ -0,0 +1,421 @@
...
+# basis*. The Bell basis is a set of four entangled states:
+#
+# .. math::
+#
+#     \begin{align*}
+#     \vert \psi_+\rangle &= \frac{1}{\sqrt{2}} \left( \vert 00\rangle + \vert 11\rangle \right), \\
+#     \vert \psi_-\rangle &= \frac{1}{\sqrt{2}} \left( \vert 00\rangle - \vert 11\rangle \right), \\
+#     \vert \phi_+\rangle &= \frac{1}{\sqrt{2}} \left( \vert 01\rangle + \vert 10\rangle \right), \\
+#     \vert \phi_-\rangle &= \frac{1}{\sqrt{2}} \left( \vert 01\rangle - \vert 10\rangle \right).
+#     \end{align*}
```
Comment (ikurecic) [inline demonstrations/tutorial_teleportation.py:184]: This part should be within step 2.

File: demonstrations/tutorial_teleportation.py [hunk 12]
```diff
@@ -0,0 +1,421 @@
...
+# :math:`\alpha\vert 0\rangle + \beta\vert 1\rangle`, with :math:`\alpha` and
+# :math:`\beta` being complex coefficients. Expanding out the terms from (1), we obtain:
+#
+# .. math::
+#
+#     \frac{1}{\sqrt{2}} ( \alpha\vert 000\rangle +
+#     \beta\vert 100\rangle + \alpha \vert 011\rangle +
+#     \beta\vert 111\rangle )
+#
+# Now let's apply a :math:`CNOT` between Alice's two qubits:
```
Comment (ikurecic) [inline demonstrations/tutorial_teleportation.py:201]:
```suggestion
# Now let's apply a CNOT between Alice's two qubits:
```

File: demonstrations/tutorial_teleportation.py [hunk 13]
```diff
@@ -0,0 +1,421 @@
...
+# .. math::
+#
+#     \beta \vert 0 \rangle + \alpha \vert 1 \rangle = X \vert \psi \rangle.
+#
+# After obtaining these results, Alice could tell Bob to simply apply an :math:`X`
+# gate to his qubit to recover the original state. Similarly, if she obtained
+# :math:`\vert 10\rangle`, she would tell him to apply a :math:`Z` gate.
+#
+# In the `"traditional" version of
+# teleportation <https://quantum.country/teleportation>`__ [#Teleportation1993]_,
```
Comment (ikurecic) [inline demonstrations/tutorial_teleportation.py:262]:
```suggestion
# quantum teleportation <https://quantum.country/teleportation>`__ [#Teleportation1993]_,
```

File: demonstrations/tutorial_teleportation.py [hunk 14]
```diff
@@ -0,0 +1,368 @@
...
+def teleport(state):
+    state_preparation(state)
+    entangle_qubits()
+    basis_rotation()
+    qml.Barrier(["A", "a"], only_visual=True)
+    measure_and_update()
+
+
+state = np.array([1 / np.sqrt(2) + 0.3j, 0.4 - 0.5j])
+_ = qml.draw_mpl(teleport, style="sketch")(state)
```
[the remark this answers is not in the document]
Comment (ikurecic) [reply demonstrations/tutorial_teleportation.py:293]: Yeah, whoops. 😅

File: demonstrations/tutorial_teleportation.py [hunk 15]
```diff
@@ -0,0 +1,421 @@
...
+    measure_and_update()
+    return qml.state()
+
+
+_ = qml.draw_mpl(teleport, style="sketch")(state)
+
+##############################################################################
+#
+# Poof! Our classical signals have been turned into :math:`CNOT` and :math:`CZ`
+# gates. This is exactly what the principle of deferred measurement gives us -
```
Comment (ikurecic) [inline demonstrations/tutorial_teleportation.py:323]:
```suggestion
# gates. This is exactly what the principle of deferred measurement gives us —
```

File: demonstrations/tutorial_teleportation.py [hunk 16]
```diff
@@ -0,0 +1,421 @@
...
+            f"Current system state: {system_state}"
+        )
+    print("State successfully teleported!")
+
+
+teleport_state(state)
+
+##############################################################################
+#
+# Conclusion
```
Comment (ikurecic) [inline demonstrations/tutorial_teleportation.py:374]: Can you only use *emph* here? Or make (some of) the emphasized segments in the rest of the file **bold**? There's a lot here, feels a bit harsh.

File: demonstrations/tutorial_teleportation.metadata.json [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+        {
+            "id": "Codebook",
+            "title": "Xanadu Quantum Codebook",
+            "authors": "C. Albornoz, G. Alonso, M. Andrenkov, P. Angara, A. Asadi, A. Ballon, S. Bapat, I. De Vlugt, O. Di Matteo, P. Finlay, A. Fumagalli, A. Gardhouse, N. Girard, A. Hayes, J. Izaac, R. Janik, T. Kalajdzievski, N. Killoran, J. Soni, D. Wakeham",
+            "year": "2021",
+            "publisher": "",
+            "journal": "",
+            "doi": "",
+            "url": "https://codebook.xanadu.ai/"
+        }
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.metadata.json:58]: the codebook homepage says "please cite us like this", but I'm happy to cite the particular node (I.15) if it makes sense

File: demonstrations/tutorial_teleportation.py [hunk 18]
```diff
@@ -0,0 +1,421 @@
+r"""
+Quantum Teleportation
+=====================
+
+.. meta::
+    :property="og:description": Transmit a quantum state to an entangled qubit.
+
+.. related::
+
+    tutorial_mbqc Measurement-based quantum computation
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:10]: sorry, I should have deleted all this metadata at the top with the new way of doing things

File: demonstrations/tutorial_teleportation.py [hunk 19]
```diff
@@ -0,0 +1,421 @@
...
+
+*Author: Matthew Silverman - Posted: 1 June 2023. Last Updated 1 June 2023.*
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:12]: just removing the whole thing 😄

File: demonstrations/tutorial_teleportation.py [hunk 20]
```diff
@@ -0,0 +1,421 @@
...
+principle of deferred measurement. Let's dive in!
+
+Suppose there are two researchers named Alice and Bob, and Alice wants to send
+her quantum state to Bob. The quantum teleportation protocol enables Alice to
+do exactly this in a very elegant manner, and it can be described in four steps:
+
+1. State Preparation - Alice initializes her qubit to the state she wishes to
+teleport.
+
+2. Entanglement - A pair of entangled qubits is created and distributed to
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:29]: I'll mention a Bell state since it fits the rest of the language, and we'll be teaching them what it is shortly

File: demonstrations/tutorial_teleportation.py [hunk 21]
```diff
@@ -0,0 +1,421 @@
...
+    :width: 75%
+
+Problem: The No-Cloning Theorem
+-------------------------------
+
+You might be wondering why we need to teleport a state at all. Can't Alice
+just make a copy of it and send the copy to Bob? It turns out that copying
+arbitrary states is *prohibited*, which you can prove using something called the
+**no-cloning theorem**. The proof is surprisingly straightforward. Suppose we
+would like to design a circuit (unitary) :math:`U` that can perform the following
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:52]: I took this from the codebook and kinda liked it as it is, which is to say we are conceptually designing a circuit, but mathematically just constructing a unitary transformation. Perhaps I could say "design a circuit (unitary transformation)"?

File: demonstrations/tutorial_teleportation.py [hunk 22]
```diff
@@ -0,0 +1,421 @@
...
+step at a time.
+
+"""
+
+##############################################################################
+#
+# 1. State preparation
+# ````````````````````
+#
+# .. figure:: ../demonstrations/teleportation/teleportation-4part.svg
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:92]: the images are placeholders for now, @ludmilaasb is helping me out and making better, more informative ones

File: demonstrations/tutorial_teleportation.py [hunk 23]
```diff
@@ -0,0 +1,421 @@
...
+#     :align: center
+#     :width: 75%
+#
+# This is where things get tricky, but also very interesting. The third step of
+# the protocol is to apply a :math:`CNOT` and a Hadamard to the first two qubits. This is
+# done prior to the measurements, and labelled "change of basis". But what basis
+# is this? Notice how these two gates are the *opposite* of what we do to create a
+# Bell state. If we run them in the opposite direction, we transform the basis
```
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:173]: I'll add a detail to step 2 introducing Bell states by name

File: demonstrations/tutorial_teleportation.py (hunk 16 above)
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:374]: for sure, I'll switch it over

File: demonstrations/tutorial_teleportation.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:293]: it's fixed in the latest pennylane, I'll change this to be against `dev` : )

File: demonstrations/tutorial_teleportation.py (hunk 18 above)
[the remark this answers is not in the document]
Comment (ikurecic) [reply demonstrations/tutorial_teleportation.py]: Yeah, you're right. I'm not entirely sure where that process is at the moment, so proceed as you were. 😁

File: demonstrations/tutorial_teleportation.py (hunk 21 above)
[the remark this answers is not in the document]
Comment (ikurecic) [reply demonstrations/tutorial_teleportation.py:46]: Hmm. I thought you were just saying "unitary circuit," as some people do. 🤔 I'd add the word, yeah.

File: demonstrations/tutorial_teleportation.py (hunk 1 above)
Comment (ludmilaasb) [reply demonstrations/tutorial_teleportation.py:35]: Something like this? 

teleport_circuit_full_circuit

Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:35]: looks awesome! I was thinking that in the intro, we should just have the component titles ("State prep", "Shared entanglement", etc.) then build up to this complete image at the end. thoughts?

File: demonstrations/tutorial_teleportation.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (timmysilv) [reply demonstrations/tutorial_teleportation.py:319]: ok it wasn't enough so I also created PennyLaneAI/pennylane#4370 and this time I confirmed locally that it does what I want : )

File: demonstrations/tutorial_teleportation.py (hunk 1 above)
Comment (ludmilaasb) [reply demonstrations/tutorial_teleportation.py:35]: teleport_circuit_full_text

Comment (ludmilaasb) [reply demonstrations/tutorial_teleportation.py:35]: teleport_circuit_full_text

**Comment by timmysilv** [comment]
update for reviewers: I did a full CI build because something seemed cached, and I've confirmed the Controlled-X/Z become proper CNOT and CZ ops respectively! All the images are updated (ty @ludmilaasb!!), and I snuck the full circuit drawing into the conclusion to summarize it all

File: demonstrations/tutorial_teleportation.metadata.json [hunk 24]
```diff
@@ -0,0 +1,75 @@
+{
+    "title": "Quantum Teleportation",
+    "authors": [
+        {
+            "id": "matthew_silverman"
+        }
+    ],
+    "dateOfPublication": "2023-09-01T00:00:00+00:00",
+    "dateOfLastModification": "2023-09-01T00:00:00+00:00",
```
Comment (KetpuntoG) [inline demonstrations/tutorial_teleportation.metadata.json:9]:
```suggestion
    "dateOfPublication": "2023-10-20T00:00:00+00:00",
    "dateOfLastModification": "2023-10-20T00:00:00+00:00",
```

File: demonstrations/tutorial_teleportation.py [hunk 25]
```diff
@@ -0,0 +1,437 @@
...
+
+
+_ = qml.draw_mpl(teleport, style="pennylane")(state)
+
+##############################################################################
+#
+# Poof! Our classical signals have been turned into CNOT and CZ gates. This is
+# exactly what the principle of deferred measurement gives us — we can apply a
+# CNOT instead of Alice calling Bob and telling him to apply an X gate, and
+# likewise for the (C)Z gate. This is incredibly useful, as it allows us to
```
Comment (KetpuntoG) [inline demonstrations/tutorial_teleportation.py:337]:
```suggestion
# likewise for the CZ gate. This is incredibly useful, as it allows us to
```

File: demonstrations/tutorial_teleportation.py [hunk 26]
```diff
@@ -0,0 +1,437 @@
...
+# .. math::
+#
+#     \frac{1}{2} \vert 00\rangle(\alpha\vert 0\rangle + \beta\vert 1\rangle) + \frac{1}{2}\vert 01\rangle (\alpha\vert 0\rangle + \beta\vert 1\rangle) + \frac{1}{2}\vert 10\rangle (\alpha\vert 0\rangle + \beta\vert 1\rangle) + \frac{1}{2}\vert 11\rangle (\alpha\vert 0\rangle + \beta\vert 1\rangle)\tag{3}
+#
+# When Alice measures her two qubits at the end, no matter which outcome she
+# gets, Bob's qubit will be in the state :math:`\alpha\vert 0\rangle + \beta \vert
+# 1\rangle`. This means that our protocol has changed the state of Bob's qubit
+# into the one Alice wished to send him, which is truly incredible!
+#
+# :func:`qml.density_matrix <pennylane.density_matrix>` will trace out and return
```
Comment (KetpuntoG) [inline demonstrations/tutorial_teleportation.py:357]: Visually it doesn't look too good to start with a lower case. Can you rephrase it so that an article or something appears before the sentence?

**Review by KetpuntoG — COMMENTED** [review]
Nice! 😊
A small change and it will be ready to merge

File: demonstrations/tutorial_teleportation.py [hunk 27]
```diff
@@ -0,0 +1,437 @@
...
+# the specified subsystem, so we use this to verify the final state received by Bob.
```
Comment (timmysilv) [inline demonstrations/tutorial_teleportation.py:358]:
```suggestion
# We can use :func:`qml.density_matrix <pennylane.density_matrix>` to trace out
# and return Bob's subsystem as a density matrix, which is a more general
# description of the state of his qubit. We will use this to verify that Alice's
# state was successfully teleported to Bob's qubit.
```

**Review by KetpuntoG — APPROVED** [review]
🚀 Awesome work!

**Review by larawatson — APPROVED** [review]
